pkgs/tools/networking/curl/default.nix

```diff
@@ -177,16 +182,23 @@ stdenv.mkDerivation rec {
     ln $out/lib/libcurl${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-gnutls${stdenv.hostPlatform.extensions.sharedLibrary}.4.4.0
   '';
 
-  passthru = {
+  passthru = let
+    useThisCurl = attr: attr.override { curl = finalAttrs.finalPackage; };
```

Comment:
@risicle https://github.com/NixOS/nixpkgs/pull/170828#issuecomment-1113568253
> The biggest problem I see with this though, is that fails to properly deal with a package's numerous variants because of the self-referential problems passthru.tests

with `finalAttrs.finalPackage` that issue has been solved

Reply:
How has this been around for over a year and I've only just learned of it??

Reply:
I merged the pr 1 month ago https://github.com/NixOS/nixpkgs/pull/119942